reflex/reflex.py

```diff
@@ -168,63 +155,29 @@ def run(
     telemetry.send(f"run-{env.value}", config.telemetry_enabled)
 
     # Run the frontend and backend.
-    if frontend:
+    if run_frontend:
         setup_frontend(Path.cwd())
+        frontend_port = processes.change_or_terminate_port(frontend_port, "frontend")
         threading.Thread(target=frontend_cmd, args=(Path.cwd(), frontend_port)).start()
-    if backend:
+    if run_backend:
+        module = f"{config.app_name}.{config.app_name}"
+        backend_port = processes.change_or_terminate_port(backend_port, "backend")
         threading.Thread(
             target=backend_cmd,
-            args=(app.__name__, backend_host, backend_port),
+            args=(module, backend_host, backend_port),
         ).start()
 
-    # Display custom message when there is a keyboard interrupt.
+    # Catch a keyboard interrupt.
     signal.signal(signal.SIGINT, processes.catch_keyboard_interrupt)
 
 
-@cli.command()
-def deploy(dry_run: bool = typer.Option(False, help="Whether to run a dry run.")):
-    """Deploy the app to the Reflex hosting service."""
-    # Get the app config.
-    config = get_config()
-
-    # Check if the deploy url is set.
-    if config.rxdeploy_url is None:
-        typer.echo("This feature is coming soon!")
-        return
-
-    # Compile the app in production mode.
-    typer.echo("Compiling production app")
-    export()
-
-    # Exit early if this is a dry run.
-    if dry_run:
-        return
-
-    # Deploy the app.
-    data = {"userId": config.username, "projectId": config.app_name}
-    original_response = httpx.get(config.rxdeploy_url, params=data)
-    response = original_response.json()
-    frontend = response["frontend_resources_url"]
-    backend = response["backend_resources_url"]
-
-    # Upload the frontend and backend.
-    with open(constants.FRONTEND_ZIP, "rb") as f:
-        httpx.put(frontend, data=f)  # type: ignore
-
-    with open(constants.BACKEND_ZIP, "rb") as f:
-        httpx.put(backend, data=f)  # type: ignore
-
-
 @cli.command()
 def export(
-    zipping: bool = typer.Option(
-        True, "--no-zip", help="Disable zip for backend and frontend exports."
```

Comment:
is there a reason `--no-zip` is going away? it's very convenient for docker container builds

Reply:
I'll add this back